### PR TITLE
Fixing compile warnings

### LINKED ISSUE #6645: An annoying compiler warning
### Description of the enhancement or error report
Recently, `Material` is enhanced with a new virtual method `void computeProperties(unsigned int)`. This however can produce a warning for a derived class, which is overwriting the previous `void computeProperties()` with a Woverloaded-virtual warning. 

### Rationale for the enhancement or information for reproducing the error
Because this is caused by the interface change in the framework, it should be framework developers who fix this warning for all impacted applications IMO.

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)
No calculations will be impacted. Also application developers can fix these warnings on their own. So this issue should be minor.

### PR DESCRIPTION
**Pull request by aeslaughter**
This fixes compiler warnings making framework, modules, and tests clean for gcc debug build on my mac.

(refs #1777)
(closes #6645)

### CODE REVIEW

**Comment by laagesen** [comment]
My mistake on the `SmoothSuperellipsoidBaseIC.C` - currently the semi-axis c isn't used in the gradient function in the ICs. I initially wrote all the functions for 2D, then I added 3D capability to the IC function, but neglected to put it in the gradient function. I will fix it by adding the 3D capability to the gradient function, sorry.

File: python/TestHarness/TestHarness.py [hunk 1]
```diff
@@ -393,6 +393,7 @@ def testOutputAndFinish(self, tester, retcode, output, start=0, end=0):
     else:
       result = 'FAILED (%s)' % reason
       did_pass = False
+
```
Comment (permcody) [inline python/TestHarness/TestHarness.py]: Awesome contribution! Love the blank line here!

File: framework/build.mk [hunk 2]
```diff
@@ -202,13 +202,17 @@ endif
...
 
 # Add method to list of defines passed to the compiler
-libmesh_CXXFLAGS       += -DMETHOD=$(METHOD)
+libmesh_CXXFLAGS += -DMETHOD=$(METHOD)
 
 # treat these warnings as errors (This doesn't seem to be necessary for Intel)
 ifneq (,$(findstring g++,$(cxx_compiler)))
-  libmesh_CXXFLAGS     += -Werror=return-type -Werror=reorder
-endif
+  libmesh_CXXFLAGS c+= -Werror=return-type -Werror=reorder
```
Comment (andrsd) [inline framework/build.mk]: what is this `c` here?

File: framework/build.mk [hunk 3]
```diff
@@ -202,13 +202,17 @@ endif
...
 
+	# Disable the long string warning from GCC
+	# warning: string length ‘524’ is greater than the length ‘509’ ISO C90 compilers are required to support [-Woverlength-strings]
+	libmesh_CXXFLAGS += -Woverlength-strings
+endif
+d
```
Comment (andrsd) [inline framework/build.mk]: What is this `d` for?

**Pull request merged**